src/pages/workspaces/workspace/analysis/AppLauncher.js

```diff
@@ -166,7 +166,11 @@ const ApplicationLauncher = _.flow(
       Utils.withBusyState(setBusy),
       withErrorReporting('Error setting up analysis file syncing')
     )(async () => {
-      const localBaseDirectory = ''
+      //The special case here is because for GCP, Jupyter and JupyterLab can both be run on the same runtime and a
+      //user may toggle back and forth between them. In order to keep notebooks tidy and in a predictable location on
+      //disk, we mirror the localBaseDirectory used by edit mode for Jupyter.
+      //Once Jupyter is phased out in favor of JupyterLab for GCP, the localBaseDirectory can be '' for all cases
+      const localBaseDirectory = !!googleProject ? `${workspaceName}/edit` : ''
```

Comment:
Will this cause problems for existing cloud environments?

Reply:
If I've written it correctly, it shouldn't. But now that I look at it more, I think it needs to handle RStudio and other apps separately from Jupyter.

Reply:
Yeah, the `AppLauncher` handles welder setup for all apps `except` Jupyter. The `AnalysisLauncher` handles setup for Jupyter because Jupyter opens to a specific analysis. For all other tools, the AppLauncher path is used and the welder call is generalized to localize all files at once.

Reply:
Thanks- this is definitely wrong then. I will follow up with another commit shortly.

Reply:
Fixed in e2a186c5b6ebc42e25f55dceea5acd3a8deb1f04

RStudio and other apps are now initiated with the correct storage links